app/models/news_item.rb

```diff
@@ -1,7 +1,10 @@
 class NewsItem < ActiveRecord::Base
   extend Enumerize
-  enumerize :state, in: [:draft, :archived, :published]
+  AVAILABLE_STATES = %w(draft archived published)
+  enumerize :state, in: AVAILABLE_STATES
   scope :published, -> { where(state: :published).order(:published_at) }
-  validates :published_at, presence: true, if: -> { state.published? } 
-  validates_presence_of :title, :state
+  validates :published_at, presence: true, if: -> { state.try(:published?) }
+  validates :title, presence: true, length: { maximum: 255 }
+  validates :state, presence: true, inclusion: { in: AVAILABLE_STATES }
+  validates :body, presence: true, length: { maximum: 4000 }
```

Comment:
I just double-checked and postgres has no limit on its text field length: http://www.postgresql.org/docs/9.1/static/datatype-character.html

I think I reflexively did that also because it's a security recommendation to limit intake of information from users (e.g. preventing users from providing endless data that would consume all server resources).

In any case, I just checked the live site posts, and although only one out of all hit the 5000 char limit, if I take their HTML content into account, they get much higher than that. 

I'll bump up that limit to 64k (65536 bytes) just to be safe.